development.md

```diff
@@ -0,0 +1,105 @@
+# The Compose Specification - Development support
+{:.no_toc}
+
+*Note:* Development support is an OPTIONAL part of the Compose Specification
+
+* ToC
+{:toc}
+
+## Introduction
+
+Compose specification is a platform-neutral way to define multi-container applications. A Compose implementation
+focussing on development use-case to run application on local machine will obviously also support some development hooks to improve developer velocity on his local workflow (also known as "inner loop"). The Compose Development specification allows to define how a Compose implementation SHOULD behave to efficiently assist the developer.
+
+## Definitions
+
+Compose Specification is extended to support an OPTIONAL `development` subsection on services. This section
+defines the development constraints and workflows to be set bu a Compose Implementation to assist the developer
+working on service codebase. Only a subset of Compose file services MAY define such a Development subsection.
+
+## Illustrative sample
+
+The following sample illustrates Compose specification concepts with a concrete sample application. The sample is non-normative.
+
+```yaml
+services:
+  frontend:
+    image: awesome/webapp
+    build: ./webapp
+    development:
+      watch: 
+        update: sync
+        path: src:/app
+        ignores:
+            - node_modules/
+
+  backend:
+    image: awesome/backend
+    build: ./backend
+    development:
+      watch: 
+        update: build
+      quiet-period: 5s
+
+  db:
+    image: awesome/database
+    development:
+      watch: sighup
+```
+
+## Developemnt mode definition
+
+The `dev` element define configuration options that are applied by Compose implementations to assist developer during development of a service.
+
+### watch
+
+`watch` can be specified either as a string containing policy or a detailed structure/
+
+```yml
+services:
+  webapp:
+    development: 
+      watch: build
+```
+is equivalent to 
+
+```yml
+services:
+  webapp:
+    development: 
+      watch: 
+        policy: build
+```
+
+
+The `watch` attribute defines `update` policy to update a running service as Compose Implementation detects 
+changes in service source code. "Source code" to watch is identified by the `build.context` section.
+
+While Compose implementations can support custom values, the specification defines:
+- `build`: rebuild service image based on the `build` section and restart the service with updated image
+- `sync`: synchronize source files with container content. `path` attribute is REQUIRED to define the mapping 
+to apply for local files within service `build.context` and container's filesystem.
+- `sighup`: send SIGHUP signal to container, to force a reload configuration/restart. This is typically used 
+for daemon services
+
+If not explicitely configured, and service has a `build` section, then `build` MUST be set as the default policy.
```

Comment:
```suggestion
If not explicitly configured, and service has a `build` section, then `build` MUST be set as the default policy.
```
typo

Reply:
This is potentially quite a big change in default behaviour. Does this mean that Compose implementations that do support `development` section will assume that all local images need to be rebuilt on each code change?

Reply:
this _tells_ compose implementation how it _could_ refresh an image by rebuilding. This is basically the "lowest common denominator". For compiled languages without incremental build capability there's no better choice. Other stacks _might_ offer smoother re-build scenario and we could then embrace those with dedicated policies.

The way a Compose Implementation exposes this to user is TO BE DEFINED by implementation (so that we don't prevent creativity :P). I assume only service image with some detected code change would be rebuilt on code change, and dependent service restarted